packages/pdf/src/RPA/PDF/keywords/document.py

```diff
@@ -0,0 +1,898 @@
+        template file.
+
+        **Examples**
+
+        **Robot Framework**
+
+        .. code-block:: robotframework
+
+            *** Settings ***
+            Library    RPA.PDF
+
+            *** Variables ***
+            ${TEMPLATE}    order.template
+            ${PDF}         result.pdf
+            &{DATA}        name=Robot Generated
+            ...            email=robot@domain.com
+            ...            zip=00100
+            ...            items=Item 1, Item 2
+
+            *** Tasks ***
+            Create PDF from HTML template
+                Template HTML to PDF   ${TEMPLATE}  ${PDF}  ${DATA}
+
+        **Python**
+
+        .. code-block:: python
+
+            from RPA.PDF import PDF
+
+            p = PDF()
+            orders = ["item 1", "item 2", "item 3"]
+            data = {
+                "name": "Robot Process",
+                "email": "robot@domain.com",
+                "zip": "00100",
+                "items": "<br/>".join(orders),
+            }
+            p.template_html_to_pdf("order.template", "order.pdf", data)
+
+        :param template: filepath to the HTML template.
+        :param output_path: filepath where to save PDF document.
+        :param variables: dictionary of variables to fill into template, defaults to {}.
+        """
+        variables = variables or {}
+
+        with open(template, "r") as templatefile:
+            html = templatefile.read()
+        for key, value in variables.items():
+            html = html.replace("{{" + key + "}}", str(value))
+
+        self.html_to_pdf(html, output_path)
+
+    @keyword
+    def html_to_pdf(
+        self,
+        content: str,
+        output_path: str,
+    ) -> None:
+        """Generate a PDF file from HTML content.
+
+        Note that input must be well-formed and valid HTML.
+
+        **Examples**
+
+        **Robot Framework**
+
+        .. code-block:: robotframework
+
+            ***Settings***
+            Library    RPA.PDF
+
+            ***Tasks***
+            Example Keyword
+                HTML to PDF    ${html_content_as_string}  /tmp/output.pdf
+
+        .. code-block:: python
+
+            from RPA.PDF import PDF
+
+            pdf = PDF()
+
+            def example_keyword():
+                pdf.html_to_pdf(html_content_as_string, "/tmp/output.pdf")
+
+        :param content: HTML content.
+        :param output_path: filepath where to save the PDF document.
+        """
+        output_path = Path(output_path) if output_path else self.default_output
+        self._write_html_to_pdf(content, output_path)
+
+    def _write_html_to_pdf(self, html: str, output_path: str) -> None:
+        self.logger.info("Writing output to file %s", output_path)
+        self.fpdf.add_page()
+        self.fpdf.write_html(html)
+        self.fpdf.output(name=output_path)
+        # self.__init__()  # TODO: what should happen here exactly?
+        self.fpdf = PDF()
```

Comment:
if a new instance is being created for each invocation of `_write_html_to_pdf()`, maybe fpdf could just be a local var in the function and not stored in the class